app/controllers/general.php

```diff
@@ -289,19 +291,66 @@
         ->noContent();
 }, ['request', 'response']);
 
-App::error(function ($error, $utopia, $request, $response, $layout, $project) {
+App::error(function ($error, $utopia, $request, $response, $layout, $project, $logger, $user, $loggerBreadcrumb) {
     /** @var Exception $error */
     /** @var Utopia\App $utopia */
     /** @var Utopia\Swoole\Request $request */
     /** @var Appwrite\Utopia\Response $response */
     /** @var Utopia\View $layout */
     /** @var Appwrite\Database\Document $project */
+    /** @var Utopia\Logger\Logger $logger */
+    /** @var Appwrite\Database\Document $user */
+
+    $version = App::getEnv('_APP_VERSION', 'UNKNOWN');
+    $route = $utopia->match($request);
+
+    if($error->getCode() >= 500 || $error->getCode() === 0) {
+        $log = new Utopia\Logger\Log();
+
+        if(!$user->isEmpty()) {
+            $log->setUser(new User($user->getId()));
+        }
+
+        $log->setNamespace("http");
+        $log->setServer(App::getEnv("_APP_LOGGING_SERVERNAME", "selfhosted-001"));
+        $log->setVersion($version);
+        $log->setType(Log::TYPE_ERROR);
+        $log->setMessage($error->getMessage());
+
+        $log->setTags([
+            'method' => $route->getMethod(),
+            'url' => $route->getPath(),
+            'code' => $error->getCode(),
+            'verbose_type' => get_class($error),
+            'projectId' => $project->getId(),
+            'hostname' => $request->getHostname(),
+            'locale' => (string)$request->getParam('locale', $request->getHeader('x-appwrite-locale', '')),
+        ]);
+
+        $log->setExtra([
```

Comment:
Same as above. If this options are only for the Sentry adapter, we can add them only for Sentry. We can also add a method that checks whether the current adapter support a specific feature. Would love to share more if this is still not clear enough.

Reply:
Updated to `addExtra`